packages/target-electron/src/ipc.ts

```diff
@@ -173,6 +174,34 @@ export async function init(cwd: string, logHandler: LogHandler) {
     ev.returnValue = app.getPath(arg)
   })
 
+  ipcMain.handle('checkMediaAccess', (_ev, mediaType: string) => {
```

Comment:
A type annotation here would be nice as well.

Reply:
This is not really possible, since the return type of systemPreferences.getMediaAccessStatus is String and not MediaAccesStatus as expected
> Returns string - Can be not-determined, granted, denied, restricted or unknown.

I couldn't find a documentation about when it returns "restricted" so I only covered the "denied" case

Reply:
In that case the `runtime.ts` declaration of `checkMediaAccess` should also return a string.

Reply:
I don't agree. The documentation says it can have one of these values but the type is string which means it is not correctly typed. Why should we keep that missing type definition.
Right now we can only handle " not-determined" and "denied" - maybe later more, but just having a string does not help in any ways I would say

Reply:
probably these values come from https://developer.apple.com/documentation/avfoundation/avauthorizationstatus#Status-Values

so restricted means "A status that indicates the app isn’t permitted to use media capture devices."

Reply:
Well then the system would just show the message "Permission denied" which makes sense since the user can't change the permission